ngraph/core/include/openvino/core/except.hpp

```diff
@@ -133,6 +171,23 @@ class OPENVINO_API AssertFailure : public Exception {
 /// \throws ::ov::AssertFailure if `cond` is false.
 #define OPENVINO_ASSERT(...) OPENVINO_ASSERT_HELPER(::ov::AssertFailure, "", __VA_ARGS__)
 
+/// \brief Macro to check whether a boolean condition holds and throw exception of specified type
+/// \param exc_class Type of exception to throw
+/// \param cond Condition to check
+/// \param ... Additional error message info to be added to the error message via the `<<`
+///            stream-insertion operator. Note that the expressions here will be evaluated lazily,
+///            i.e., only if the `cond` evalutes to `false`.
+/// \throws ::ov::exc_class if `cond` is false.
+#define OPENVINO_ASSERT_THROW(exc_class, ...) \
+    OPENVINO_ASSERT_HELPER(::ov::exc_class, OPENVINO_FUNCTION_NAME, __VA_ARGS__)
+
+/// \brief Throw exception of specified type
+/// \param exc_class Type of exception to throw
+/// \param ... Additional error message info to be added to the error message via the `<<`
+///            stream-insertion operator. Note that the expressions here will always be evaluated
+/// \throws ::ov::exc_class
+#define OPENVINO_THROW(exc_class, ...) OPENVINO_ASSERT_HELPER1(::ov::exc_class, OPENVINO_FUNCTION_NAME, false)
```

Comment:
do we need specializations in new API? I suppose generic ov::Exception can be enough. 
Or at least we need to move only required specializations, not all of them.